HW3/hw03.Rmd

```diff
@@ -0,0 +1,270 @@
+---
+title: "HW3"
+author: "Roxanne Felig"
+date: "2/8/2021"
+output:
+    html_document:
+        theme: paper
+    github_document: default
+---
+# Homework 03
+
+You will analyze data looking at the relationship between **green reputation** and three personality traits--**compassion**, **intellectual curiosity**, and **openness to experiences**.
+The dataset includes data from **students** and **non-students**.
+
+```{r}
+dictionary <- readr::read_csv("/Users/roxannefelig/Desktop/progdata/hw-roxannefelig/HW3/green_dictionary.csv")
+green_data <- readr::read_csv("/Users/roxannefelig/Desktop/progdata/hw-roxannefelig/HW3/green_data.csv")
+
+library(tidyverse)
+library(dplyr)
+library(psych)
+library(naniar)
+library(reshape2)
+library(scales)
+```
+
+```{r view files}
+View(dictionary)
+View(green_data)
+
+```
+
+
+```{r question 1}
+###1. Inspect the item responses (e.g., with graphs or by summarizing distinct values). Is anything unusual?
+
+str(green_data)
+summary(green_data) #I notice that some missing values are coded as NA while others are coded as -99, so I'm going to re-code the -99s as NAs.
+
+data<-green_data %>% 
+  mutate_at(vars(comp1:student),
+         ~ifelse(. == -99, NA, .)
+  )
+
+summary(data) #now we see that the min values for variables are 1, not -99.
+
+melt_data<-melt(data) #collapses data to one column
+View(melt_data)
+
+#plotting using the melted data and the facet wrap will allow us to see histograms for each scale item, and the frequency of each response choice.
+
+ggplot(melt_data)+
+  aes(x=value)+
+  geom_histogram()+
+  facet_wrap(~variable, 
+             scales="free")
+```
+
+
+```{r question 2}
+### 2. Compute total scores for the four scales. Recode variables as needed.
+
+#First, we need to reverse code items as indicated in the dictionary. The reverse coded items are green2,comp6:comp10, intel7:intel10,and open7:open10. 
+#two ways to reverse code shown below:
+
+reversed1<- c("green2", "comp6", "comp7","comp8","comp9","comp10","intel7",
+             "intel8","intel9","intel10","open7","open8","open9","open10")
+
+reversed2 <- dictionary %>%
+  filter(Keying == -1) %>%
+  pull(Item)     
+     
+data %>%
+  mutate(
+    across(all_of(reversed2),
+           ~ recode(.x, "5" = 1, "4" = 2, "3" = 3, "2" = 4, "1" = 5),
+           .names = "{.col}r")
+  )
+
+#total scores
+summary(data)
+View(data)
+
+#we will use the row wise function for this so that we can create total scores for each person.
+
+new_data<-data %>%
+  rowwise() %>% 
+  mutate(
+    .id = id,
+    green_total = mean(c_across(green1:green5), na.rm = TRUE),
+    comp_total = mean(c_across(comp1:comp10), na.rm = TRUE),
+    intel_total = mean(c_across(intel1:intel10), na.rm = TRUE),
+    open_total = mean(c_across(open1:open10), na.rm = TRUE),
+    .before = everything()
+  ) 
+
+summary(new_data)
```

Comment:
I used View(new_data) and found some items coded as "NaN" and some as "NA", you may want to double check with it.

Reply:
I think that gave me an error when I try to run your code for question 3, maybe it's only on my end though.

Reply:
huh, that's weird.. I ran into no errors and had no problem knitting. Do you remember which line the error was on?

Reply:
I tried to run it again today and there was no error lol.